### ISSUE TITLE
numpy.clip does not respect out array order

### ISSUE THREAD

**Issue by llchan** [numpy/numpy#7633]
It appears that numpy.clip does not respect the out array order:

``` python
>>> import numpy as np
>>> np.__version__
'1.11.0'
>>> A = np.arange(15).reshape(5, 3)
>>> A
array([[ 0,  1,  2],
       [ 3,  4,  5],
       [ 6,  7,  8],
       [ 9, 10, 11],
       [12, 13, 14]])
>>> Af = np.zeros_like(A, order='F')
>>> Af
array([[0, 0, 0],
       [0, 0, 0],
       [0, 0, 0],
       [0, 0, 0],
       [0, 0, 0]])
>>> np.clip(A, 0, 7, out=Af)  # wrong
array([[0, 5, 7],
       [1, 6, 7],
       [2, 7, 7],
       [3, 7, 7],
       [4, 7, 7]])
>>> np.clip(A, 0, 7)  # expected
array([[0, 1, 2],
       [3, 4, 5],
       [6, 7, 7],
       [7, 7, 7],
       [7, 7, 7]])
```

**Comment by jaimefrio** [comment]
The check [here](https://github.com/numpy/numpy/blob/maintenance/1.11.x/numpy/core/src/multiarray/calculation.c#L1107) should also verify that both the input and output arrays have the same order (C or Fortran). I think this code could be greatly simplified by using the new iterator.

**Comment by PiyushGoyal443** [comment]
Can I work on this?

**Comment by mattip** [comment]
Closed by #9209
